File: app/src/main/java/ds/features/model/CurrWeatherData.java
package ds.features.model;

import java.util.List;

public final class CurrWeatherData {

	public Coord coord;
	public List<Weather> weather;
	public String base;
	public Main main;
	public Wind wind;
	public Clouds clouds;
	public long dt;
	public Sys sys;
	public long id;
	public String name;
	public long cod;


	public static final class Coord {

		public double lon;
		public double lat;
	}


	public static final class Weather {

		public long id;
		public String main;
		public String description;
		public String icon;

	}


	public static final class Main {

		public double temp;
		public long pressure;
		public long humidity;
		public double temp_min;
		public double temp_max;


	}


	public static final class Wind {

		public double speed;
		public float deg;

	}


	public static final class Clouds {

		public long all;

	}


	public static final class Sys {

		public long type;
		public long id;
		public double message;
		public String country;
		public long sunrise;
		public long sunset;


	}
}

File: app/src/main/java/ds/features/model/CurrWeatherData.kt
package ds.features.model

data class CurrWeatherData(var coord: CurrWeatherData.Coord,
                           var weather: List<CurrWeatherData.Weather>,
                           var base: String,
                           var main: CurrWeatherData.Main,
                           var wind: CurrWeatherData.Wind,
                           var clouds: CurrWeatherData.Clouds,
                           var dt: Long,
                           var sys: CurrWeatherData.Sys,
                           var id: Long,
                           var name: String,
                           var cod: Long) {




    data class Coord(var lon: Double,
                     var lat: Double)


    data class Weather(var id: Long,
                       var main: String,
                       var description: String,
                       var icon: String)

    data class Main(var temp: Double,
                    var pressure: Long,
                    var humidity: Long,
                    var temp_min: Double)

    data class Wind(var speed: Double,
                    var deg: Float = 0.toFloat())


    data class Clouds(var all: Long)


    data class Sys(
            var type: Long,
            var id: Long,
            var message: Double,
            var country: String,
            var sunrise: Long,
            var sunset: Long
    )


}